### PR TITLE
dibs: I will implement exercise hello world

### PR DESCRIPTION
**Pull request by dvberkel**
This is work in progress

### CODE REVIEW

**Comment by dvberkel** [comment]
This can be reviewed

File: config.json [hunk 1]
```diff
@@ -28,6 +29,11 @@
   ],
   "exercises": [
     {
+      "slug": "hello-world",
+      "difficulty": 1,
+      "topics": []
```
Comment (stevejb71) [inline config.json]: Difficulty is set to the same as Bob, which I think is a fair bit harder.

File: exercises/hello-world/example.ml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+let greet subject =
+  match subject with
+  | None    -> "Hello, World!"
+  | Some(s) -> String.concat "" ["Hello, "; s; "!"]
```
Comment (stevejb71) [inline exercises/hello-world/example.ml]: You can use ^ for concatenation of strings

File: config.json (hunk 1 above)
Comment (dvberkel) [inline config.json]: All the difficulties are set to 1, which is the default. They are in the process of being set correctly, but it takes some time

File: exercises/hello-world/example.ml (hunk 2 above)
Comment (dvberkel) [inline exercises/hello-world/example.ml]: Great! I searched for a concatenation operator, but I missed it. I will change the implementation.

**Comment by dvberkel** [comment]
I have no problem with introducing Option type in the `hello-world` exercise. It is idiomatic OCaml and characteristic for the language. 

Furthermore, other languages with an option type use it as well. Look for example at [elm](https://github.com/exercism/xelm/blob/master/exercises/hello-world/HelloWorld.example).

 Having said that, it does seem fair to explain the use of options in a `HINT.md`.

**Comment by kytrinyx** [comment]
> it does seem fair to explain the use of options in a HINTS.md

Yeah, agreed—if it's someone's first meeting with OCaml they might have no idea where to look to find out what's idiomatic.

**Pull request merged**